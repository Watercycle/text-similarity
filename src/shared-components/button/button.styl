@import '~/src/shared-styles/colors'

.root
    color: $primary
