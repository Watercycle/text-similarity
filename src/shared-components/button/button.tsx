import { Component } from 'solid-js'
import styles from './button.styl'

export interface Props {}

export const WcButton: Component<Props> = props =>
  <button class="btn ripple">{props?.children}</button>
