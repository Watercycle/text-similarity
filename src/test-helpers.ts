import * as dom from 'solid-js/dom'

export function render(jsx: JSX.Element): string {
  const div = document.createElement('div')

  const dispose = dom.render(() => jsx, div)
  dispose()

  return div.children[0].outerHTML;
}

export function assertMatchesSnapshot(jsx: JSX.Element) {
  const html = render(jsx);
  expect(html).toMatchSnapshot()
}
