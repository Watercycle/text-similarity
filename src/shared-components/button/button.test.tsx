import { assertMatchesSnapshot } from '../../test-helpers'
import { WcButton } from './button'

describe('WcButton', () => {
    it('renders', () => {
        assertMatchesSnapshot(() => <WcButton>Yeet</WcButton>)
    })
});
