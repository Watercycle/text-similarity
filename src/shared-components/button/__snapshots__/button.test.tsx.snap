// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WcButton renders 1`] = `"<button class=\\"btn ripple\\">Yeet</button>"`;
